apis/apps/v1alpha1/cluster_types.go

```diff
@@ -306,6 +305,81 @@ type UserResourceRefs struct {
 	ConfigMapRefs []ConfigMapRef `json:"configMapRefs,omitempty"`
 }
 
+// InstanceTemplate defines values to override in pod template.
+type InstanceTemplate struct {
+	// Number of replicas of this template.
+	// Default is 1.
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Defines the name of the instance.
+	// Only applied when Replicas is 1.
+	// +optional
+	Name *string `json:"name,omitempty"`
+
+	// GenerateName is an optional prefix, used by the server, to generate a unique
+	// name ONLY IF the Name field has not been provided.
+	// If this field is used, the name returned to the client will be different
+	// than the name passed. This value will also be combined with a unique suffix.
+	// The provided value has the same validation rules as the Name field,
+	// and may be truncated by the length of the suffix required to make the value
+	// unique on the server.
+	//
+	// Applied only if Name is not specified.
+	// +optional
+	GenerateName *string `json:"generateName,omitempty"`
```

Comment:
All the XXName filed should add the validation of length

Reply:
fixed: [2eacb23](https://github.com/apecloud/kubeblocks/pull/6776/commits/2eacb23d488df6bb364c72c7919ebe23f1258d7d)